diff.go

```diff
@@ -316,16 +361,17 @@ func (d *BundleDiff) Empty() bool {
 
 // ApplicationDiff stores differences between an application in a bundle and a model.
 type ApplicationDiff struct {
-	Missing     DiffSide              `yaml:"missing,omitempty"`
-	Charm       *StringDiff           `yaml:"charm,omitempty"`
-	Series      *StringDiff           `yaml:"series,omitempty"`
-	Placement   *StringDiff           `yaml:"placement,omitempty"`
-	NumUnits    *IntDiff              `yaml:"num_units,omitempty"`
-	Scale       *IntDiff              `yaml:"scale,omitempty"`
-	Expose      *BoolDiff             `yaml:"expose,omitempty"`
-	Options     map[string]OptionDiff `yaml:"options,omitempty"`
-	Annotations map[string]StringDiff `yaml:"annotations,omitempty"`
-	Constraints *StringDiff           `yaml:"constraints,omitempty"`
+	Missing          DiffSide                       `yaml:"missing,omitempty"`
+	Charm            *StringDiff                    `yaml:"charm,omitempty"`
+	Series           *StringDiff                    `yaml:"series,omitempty"`
+	Placement        *StringDiff                    `yaml:"placement,omitempty"`
+	NumUnits         *IntDiff                       `yaml:"num_units,omitempty"`
+	Scale            *IntDiff                       `yaml:"scale,omitempty"`
+	Expose           *BoolDiff                      `yaml:"expose,omitempty"`
+	ExposedEndpoints map[string]ExposedEndpointDiff `yaml:"exposed-endpoints,omitempty"`
```

Comment:
`num_units` uses underscore and you use `-` (note I prefer the readability of `-`), we should standardise though